modules/multisignatures.js

```diff
@@ -80,158 +80,221 @@ function Multisignatures(cb, scope) {
+			);
+			library.bus.message('signature', signature, true);
+			return setImmediate(cb);
+		}
+	);
+};
 
-			if (!account) {
-				return setImmediate(cb, 'Account not found');
-			}
+/**
+ * Main function for processing received signature, includes:
+ * - multisignature account creation
+ * - send from multisignature account
+ *
+ * @public
+ * @param {Object} signature - Signature data
+ * @param {string} [signature.publicKey] - Public key of account that created the signature (optional)
+ * @param {string} signature.transactionId - Id of transaction that signature was created for
+ * @param {string} signature.signature - Actual signature
+ * @param {function} cb - Callback function
+ * @implements {library.balancesSequence.add} - All processing here is done through balancesSequence
+ * @returns {setImmediateCallback} cb, err
+ */
+Multisignatures.prototype.processSignature = function(signature, cb) {
+	if (!signature) {
+		const message = 'Unable to process signature, signature not provided';
+		library.logger.error(message);
+		return setImmediate(cb, message);
+	}
 
-			multisignatureTransaction.signatures =
-				multisignatureTransaction.signatures || [];
+	// From now perform all the operations via balanceSequence
+	library.balancesSequence.add(balanceSequenceCb => {
+		// Grab transaction with corresponding ID from transaction pool
+		const transaction = modules.transactions.getMultisignatureTransaction(
+			signature.transactionId
+		);
 
-			if (
-				multisignatureTransaction.signatures.indexOf(transaction.signature) >= 0
-			) {
-				return setImmediate(cb, 'Signature already exists');
-			}
+		if (!transaction) {
+			const message =
+				'Unable to process signature, corresponding transaction not found';
+			library.logger.error(message, { signature });
+			return setImmediate(balanceSequenceCb, message);
+		}
 
-			try {
-				for (let i = 0; i < multisignatures.length && !verify; i++) {
-					verify = library.logic.transaction.verifySignature(
-						multisignatureTransaction,
-						multisignatures[i],
-						transaction.signature
-					);
-				}
-			} catch (e) {
-				library.logger.error(e.stack);
-				return setImmediate(cb, 'Failed to verify signature');
-			}
+		// If there are no signatures yet - initialise with empty array
+		transaction.signatures = transaction.signatures || [];
 
-			if (!verify) {
-				return setImmediate(cb, 'Failed to verify signature');
-			}
+		// Check if received signature already exists in transaction
+		if (transaction.signatures.includes(signature.signature)) {
+			const message = 'Unable to process signature, signature already exists';
+			library.logger.error(message, { signature, transaction });
+			return setImmediate(balanceSequenceCb, message);
+		}
 
-			library.network.io.sockets.emit(
-				'multisignatures/signature/change',
-				multisignatureTransaction
+		// Process signature for multisignature account creation transaction
+		if (transaction.type === transactionTypes.MULTI) {
```

Comment:
I believe this check should be in start of function and return if its not a multi-sig transaction.

Reply:
I don't understand, can you clarity? We're doing the check for multisignature transaction here - https://github.com/LiskHQ/lisk/pull/2275/files#diff-7d87322fa8c7f1c80db1ecbc41cc0bb5R265